vmdb/app/models/ems_refresh/parsers/foreman.rb

```diff
@@ -0,0 +1,88 @@
+module EmsRefresh::Parsers
+  class Foreman
+    def self.ems_inv_to_hashes(inv)
+      new.ems_inv_to_hashes(inv)
+    end
+
+    def ems_inv_to_hashes(inv)
+      result = {}
+      ids = {}
+      result[:customization_scripts] = customization_scripts_inv_to_hash(inv[:media], inv[:ptables])
+      add_ids(ids, result[:customization_scripts])
+      result[:operating_system_flavors] = operating_system_flavors_inv_to_hashes(inv[:operating_systems], ids)
+      add_ids(ids, result[:operating_system_flavors])
+      result[:configuration_profiles] = configuration_profile_inv_to_hashes(inv[:hostgroups], ids)
+      add_ids(ids, result[:configuration_profiles])
+      result[:configured_systems] = configured_system_inv_to_hashes(inv[:hosts], ids)
+      result
```

Comment:
Forgot to mention I was hoping this could use the `@data` and `@data_index` concept from cloud refreshers, but oh well.  That pattern is what we are considering for rewriting parsers into using a builder-like syntax.